src/encoder.js

```diff
@@ -57,6 +59,64 @@ class Encoder {
     return Web3Utils.padLeft(hexAddr, PADDED_BYTES);
   }
 
+  /*
+   * Converts a boolean to hex padded-left to 32 bytes. Accepts it in true/false or 1/0 format.
+   * @param value The boolean to convert.
+   * @return The converted boolean to padded-left hex string.
+   */
+  static boolToHex(value) {
+    if (_.isUndefined(value)) {
+      throw new Error(`value should not be undefined`);
+    }
+
+    return this.uintToHex(value ? 1 : 0);
+  }
+
+  /*
+   * Converts an int to hex padded-left to 32 bytes. 
+   * Accepts the following formats:
+   *    decimal: 12345
+   *    string: '-12345'
+   *    BN.js: <BN: 3039>
+   * @param num The number to convert.
+   * @return The converted int to padded-left hex string.
+   */
+  static intToHex(num) {
+    if (_.isUndefined(num)) {
+      throw new Error(`num should not be undefined`);
+    }
+
+    // Must be converted to Two's Complement representation to handle negative numbers
+    const twosComp = new BN(num).toTwos(256).toJSON();
+    if (_.indexOf(num.toString()) === -1) {
```

Comment:
`if (num >= 0)`?

Reply:
fixed. good catch, the logic was wrong. I wasn't looking for the `-` sign. basically positive numbers get padded with `0` and negatives with `f` in Two's Complement.